azure/auto-cidr/variables.tf

```diff
@@ -0,0 +1,11 @@
+variable "network_mask" {
+  # this tripped me up for a bit
+  # this has to be a number, required by the resource
+  # in some places we express this as a string prefixed with a slash: "/22"
+  type    = number
+  default = 22
```

Comment:
We should probably drop the default here. I don't think there is a way for us to default to a reasonable mask. Its very contextual. `/28` is common for the Azure "service" subnet, but it would be way to small for something like DataBricks or an actual compute subnet.

Reply:
It's contextual, yes, but is a sane default possible? That's why I thought `/22` was okay. For a "service" subnet, it can actually be as big `/24`. This is how the [app-serivce](https://github.com/massdriver-cloud/terraform-modules/blob/main/massdriver-application-azure-app-service/subnet.tf#L17-L27) and function-apps work. 

Flex DBs, CosmoDB are similar, but since they don't autoscale they don't need the `/24` in the "worst case". They only need at `/28`.  A `/22` allows us to make a max of ~256 give or take some peering / etc IPs I'm not accounting for. Conservatively, 250 `/22`s. 

I was thinking about a 3rd-party using this module. Specifying the least possible with a range big enough to handle services (app-service, function-app) and small enough to not be too  big.

Reply:
> is a sane default possible?
I would say no, just because it is so contextual. What is sane for a postgresql-flexible-server subnet (`/28`) isn't sane for data bricks or app-service or even a compute subnet. A `/22` subnet is 64 times larger than what's needed for postgres.

We can't make this assumption at this level. We can encode assumptions where it's used.